Marlin/src/module/tool_change.cpp

```diff
@@ -476,6 +480,12 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
       #endif
 
       feedrate_mm_s = old_feedrate_mm_s;
+      #if ENABLED(DUAL_X_CARRIAGE) && ENABLED(MIN_SOFTWARE_ENDSTOPS)
+        soft_endstop_min[X_AXIS] = active_extruder ? X2_MIN_POS : X1_MIN_POS;
+      #endif
+      #if ENABLED(DUAL_X_CARRIAGE) && ENABLED(MAX_SOFTWARE_ENDSTOPS)
+        soft_endstop_max[X_AXIS] = active_extruder ? X2_MAX_POS : X1_MAX_POS;
+      #endif
```

Comment:
Marlin knows how to set the endstops for Dual X in the `update_software_endstops` function, so that should be called here instead of adding extra logic here in `tool_change`.